### ISSUE TITLE
Snake / Kebab Case

### ISSUE THREAD

**Issue by hazmat345** [loganasherjones/yapconf#7]
* Yapcon version: 0.1.1
* Python version: 3.6.4
* Operating System: Centos 7

Say I have a config option `foo_bar`. If I want to set it on the command line it looks like this:

``--foo_bar value``

It's worse if it's a boolean option:

``--no-foo_bar``

I'd like to be able to treat the two cases `--no-foo-bar` and `--no-foo_bar` the same. If I were to just change the option name to `foo-bar` I think I'd hit the opposite problem if I tried to use environment variables. 

The only potential issue I see is if I were to have `foo-bar` *and* `foo_bar` as distinct spec options - obviously that would break. So that would need to be explicitly disallowed.

Thoughts?

**Comment by loganasherjones** [comment]
Hmm. So, I wouldn't mind converting the key `foo_bar` to `--foo-bar` on the command line. I agree that looks more command-line ish. I can support this for both snake_case and camelCase conversions. It will be weird if someone has a space in key name, but I'm going to leave that as a problem for them.

I'm less sold on adding both `--foo_bar` _and_ `--foo-bar`. If I enabled this behavior, and then you set `FOO-BAR` in the environment, it feels like I should load that too. Also, if I did it for camelCase, should I load `FOOBAR`, `FOO_BAR` or `FOO-BAR` first?

**Comment by hazmat345** [comment]
I don't think you can have an environment variable with `-` in the name. I just tried it in `zsh` and it yelled at me. Maybe you can on Windows though?

I think adding CamelCase to the equation complicates things a little too much.

Fundamentally, I'm arguing that `foo-bar` and `foo_bar` should be treated as being the 'same' option. Admittedly, that's a little unexpected. But I think the justification is pretty good:

- You can potentially load this value from the environment, where you can't have hyphens.
- You can also potentially load from CLI, where you *can* have underscores, but they're kind of weird and unexpected there.
- To resolve this, as a rule, `-` and `_` are treated as being the same for the purposes of spec names. Therefore you're forbidden from having names that differ only by `-` vs `_`.

I'm not sure I'm OK with extending that argument to CamelCase. I don't think I'm necessarily opposed, but I don't have a good idea how to handle the environment variable ordering question you raised. We could disallow having `foo-bar` and `fooBar` from a spec validation perspective, but that seems a little weirder to me.

I also don't have an answer for the question of what to do when you have spec `foo-bar` and then you have `FOO_BAR` and `fooBar` in the environment (I just checked and that's a valid thing that can happen. But I feel kind of gross using lowercase in an environment variable).

So yeah. I think what I'm asking for can be summed up by saying I'd like to always normalize `-` and `_` to the same thing. I don't have a strong opinion on the camelCase but as of now I think the complications would outweigh the benefits.

**Comment by hazmat345** [comment]
Just to document what we talked about:

It'll be better to take the opinionated approach and say that CLI will be in kebab case, while environment variables will be in snake case.
